src/components/playback/skipsegment.ts

```diff
@@ -0,0 +1,176 @@
+    if (hideTimeout) {
+        clearTimeout(hideTimeout);
+        hideTimeout = null;
+    }
+}
+
+function onHideComplete() {
+    if (skipElement) {
+        skipElement.classList.add('hide');
+    }
+}
+
+function showSkipButton(options: ShowOptions) {
+    const elem = skipElement;
+    if (elem) {
+        clearHideTimeout();
+        dom.removeEventListener(elem, dom.whichTransitionEvent(), onHideComplete, {
+            once: true
+        });
+        elem.classList.remove('hide');
+        if (!options.animate) {
+            elem.classList.add('no-transition');
+        } else {
+            elem.classList.remove('no-transition');
+        }
+
+        void elem.offsetWidth;
+
+        requestAnimationFrame(() => {
+            elem.classList.remove('skip-button-hidden');
+
+            if (!options.keep) {
+                hideTimeout = setTimeout(hideSkipButton, 6000);
+            }
+        });
+    }
+}
+
+function hideSkipButton() {
+    const elem = skipElement;
+    if (elem) {
+        elem.classList.remove('no-transition');
+        void elem.offsetWidth;
+
+        requestAnimationFrame(function() {
+            elem.classList.add('skip-button-hidden');
+
+            dom.addEventListener(elem, dom.whichTransitionEvent(), onHideComplete, {
+                once: true
+            });
+        });
+    }
+}
+
+function onPromptSkip(_e: Event, segment: MediaSegmentDto) {
+    if (!currentSegment) {
+        currentSegment = segment;
+
+        createSkipElement();
+
+        setButtonText();
+
+        showSkipButton({ animate: true });
+    }
+}
+
+function onOsdChanged(_e: Event, isOpen: boolean) {
+    if (currentSegment) {
+        if (isOpen) {
+            showSkipButton({
+                animate: false,
+                keep: true
+            });
+        } else if (!hideTimeout) {
+            hideSkipButton();
+        }
+    }
+}
+
+function onTimeUpdate() {
+    if (currentSegment) {
+        const time = playbackManager.currentTime(currentPlayer) * TICKS_PER_MILLISECOND;
+
+        if (!isInSegment(currentSegment, time)) {
+            currentSegment = null;
+            hideSkipButton();
+        }
+    }
+}
+
+function cleanupPlayer() {
+    Events.off(currentPlayer, 'promptskip', onPromptSkip);
+    Events.off(document, EventType.SHOW_VIDEO_OSD, onOsdChanged);
+    Events.off(currentPlayer, 'playbackstop', onPlaybackStopped);
+    Events.off(currentPlayer, 'timeupdate', onTimeUpdate);
+}
+
+function onPlaybackStopped() {
+    currentSegment = null;
+    hideSkipButton();
+    cleanupPlayer();
+}
+
+function bindToPlayer(player: Plugin) {
+    if (!player) {
+        return;
+    }
+
+    currentPlayer = player;
+
+    Events.on(player, 'promptskip', onPromptSkip);
+    Events.on(document, EventType.SHOW_VIDEO_OSD, onOsdChanged);
+    Events.on(player, 'playbackstop', onPlaybackStopped);
+    Events.on(player, 'timeupdate', onTimeUpdate);
+}
+
+Events.on(playbackManager, 'playerchange', (_e, newPlayer) => {
```

Comment:
I would make this a `PlaybackSubscriber` so the event handling is managed for you. 😉

Reply:
I was actually considering this but didn't look into it enough to see if that could work.

I'll change this to use PlaybackSubscriber, thanks!